user/caribou/module/src/DSO9254AEventConverter.cc

```diff
@@ -0,0 +1,377 @@
+#include "CaribouEvent2StdEventConverter.hh"
+#include "utils/log.hpp"
+#include "CLICTDFrameDecoder.hpp"
+#include "TFile.h"
+#include "TH1D.h"
+#include "time.h"
+
+using namespace eudaq;
+
+namespace{
+  auto dummy0 = eudaq::Factory<eudaq::StdEventConverter>::
+  Register<DSO9254AEvent2StdEventConverter>(DSO9254AEvent2StdEventConverter::m_id_factory);
+}
+
+uint64_t timeConverter( std::string date, std::string time );
+
+bool DSO9254AEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::StandardEventSP d2, eudaq::ConfigurationSPC conf) const{
+  
+
+  auto ev = std::dynamic_pointer_cast<const eudaq::RawEvent>(d1);
+  // No event
+  if(!ev) {
+    return false;
+  }
+
+
+  // load parameters from config file
+  double pedStartTime = conf->Get("pedStartTime", 0 ); // integration windows in [ns]
```

Comment:
You need to explicitly check here if the `config` is not `nullptr` otherwise this converter segfaults when called without a valid `config` object.

Reply:
Furthermore, these values are parsed agan and again for *every single event*. Maybe consider making them static member variables and to initialize them only once (have a look at the other Caribou converters that do something like this already).

Reply:
> You need to explicitly check here if the `config` is not `nullptr` otherwise this converter segfaults when called without a valid `config` object.

Well `corry` aborts anyway if no config file is defined. `euCliReader` still starts but falls back to the default values (0 in all cases), so it does not produce a root file. The same is true for corry if the keys are not defined in the config file. Still I think think this behavior is not ideal. I will implement some sort of error message.

Reply:
> Furthermore, these values are parsed agan and again for _every single event_. Maybe consider making them static member variables and to initialize them only once (have a look at the other Caribou converters that do something like this already).

Implemented. Although I don't get why these should be static.

Reply:
The way these decoders are implemented in EUDAQ2, there is no guarantee that the *same decoder object* will be used for consecutive events. It might as well be that a new object is created. In that case, suddenly all your variables are uninitialized instead of having the previously-assigned values from the configuration.

By making them static, the variable's scope is global and outside of the decoder objects. This means, any instance of this decoder will use the same values that you have assigned.

It's crooked but that's what the design of the decoder interface requires... :/

Does this make sense?